docs/ExtensibleMaps.md

```diff
@@ -0,0 +1,149 @@
+# Introduction
+
+Extensible maps are custom map types that can be implemented by eBPF extensions (e.g. BPF_MAP_TYPE_XSKMAP) through a new
+NMR (Network Module Registrar) provider interface. This document contains the design for adding support for extensible
+maps in eBPF-for-Windows. Extensible maps enable extensions to register and manage their own map types beyond those
+provided by the core eBPF runtime.
+
+## NMR Interface for Extensions
+A new NMR interface is added and extensions register as map info providers using NMR, similar to existing program and
+hook providers.
+
+**New NMR Provider Interface**: `EBPF_MAP_INFO_EXTENSION_IID`
+
+**Provider Registration Data**:
+```c
+typedef struct _ebpf_map_provider_data {
+    ebpf_extension_header_t header;
+    uint32_t map_type;  // Single map type per provider
+    ebpf_map_provider_dispatch_table_t* dispatch_table;
+} ebpf_map_provider_data_t;
+```
+
+**Provider Dispatch Table**:
+```c
+typedef struct _ebpf_map_provider_dispatch_table {
+    ebpf_extension_header_t header;
+    _Notnull_ ebpf_map_create_t create_map_function;
+    _Notnull_ ebpf_map_delete_t delete_map_function;
+    _Notnull_ ebpf_map_associate_program_type_t associate_program_function;
+    ebpf_map_find_element_t find_element_function;
+    ebpf_map_update_element_t update_element_function;
+    ebpf_map_delete_element_t delete_element_function;
+    ebpf_map_get_next_key_and_value_t get_next_key_and_value_function;
+} ebpf_map_provider_dispatch_table_t;
+```
+
+**Client Services** (provided by eBPF core):
+```c
+typedef struct _ebpf_map_client_dispatch_table {
+    ebpf_extension_header_t header;
+    epoch_allocate_with_tag_t epoch_allocate_with_tag;
+    epoch_allocate_cache_aligned_with_tag_t epoch_allocate_cache_aligned_with_tag;
+    epoch_free_t epoch_free;
+    epoch_free_cache_aligned_t epoch_free_cache_aligned;
+} ebpf_map_client_dispatch_table_t;
+```
+
+## Map Type Enum Partitioning
+Currently map type IDs are allocated from a global namespace. With extensible maps, global map type ID space is
+partitioned into 2 disjoint sets: for global maps (implemented in eBPFCore) and for the extensible maps.
+Global maps will use IDs from 1 to 4095. Extensible maps will use IDs 4096 onwards.
+
+- **Global Maps (1-4095)**: Reserved for core eBPF runtime map types (hash, array, etc.)
+- **Extensible Maps (4096+)**: Available for extension-implemented custom map types
+
+Extensions *can* reserve unique map type IDs by submitting PRs to update the enum in the eBPF repository.
```

Comment:
Why is "*can*"highlighted?  Just say "must".   But I don't understand any benefit to partitioning the space.  Please elaborate on what the benefit is.   (I don't think allowing collisions is a benefit or a good thing, and requiring a PR is the only way you've defined in this document to avoid it.)